### PR TITLE
RabbitEvents Event package. Initial commit

### PR DESCRIPTION
**Pull request by masterjus**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  "type": "library",
+  "license": "MIT",
+  "authors": [
+    {
+      "name": "Eugene Kirdzei",
+      "email": "masterjus@gmail.com"
+    }
+  ],
+  "require": {
+    "php": "^8.0|^8.1",
```
Comment (andrew-nuwber) [inline composer.json]: `^8.0` already includes `^8.1`

File: coverage.xml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (andrew-nuwber) [inline coverage.xml]: gitignore?

File: phpunit.xml.dist [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (andrew-nuwber) [inline phpunit.xml.dist:1]: gitignore?

File: src/Commands/stubs/observer.stub [hunk 4]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace DummyNamespace;
+
+use NamespacedDummyModel;
+
+class DummyClass
```
Comment (andrew-nuwber) [inline src/Commands/stubs/observer.stub:7]: Do you need this?

File: src/EventServiceProvider.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+use RabbitEvents\Event\Commands\ObserverMakeCommand;
+use RabbitEvents\Foundation\Context;
+
+class EventServiceProvider extends ServiceProvider
+{
+    public function boot(): void
+    {
+        $this->app->singleton(
+            Publisher::class,
+            fn($app) => new Publisher(new MessageFactory($app[Context::class]->getTransport()))
```
Comment (andrew-nuwber) [inline src/EventServiceProvider.php]:
```suggestion
            static fn($app) => new Publisher(new MessageFactory($app[Context::class]->getTransport()))
```

File: src/MessageFactory.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+use RabbitEvents\Foundation\Message;
+
+class MessageFactory
+{
+    public function __construct(private Transport $transport)
+    {
+    }
+
+    /**
+     * @param ShouldPublish $event
```
Comment (andrew-nuwber) [inline src/MessageFactory.php]: Notice: I'm not sure about adding phpdoc like that because all data already typed

**Comment by andrew-nuwber** [comment]
`src/.DS_Store` also should be gitignored

File: coverage.xml (hunk 2 above)
Comment (masterjus) [reply coverage.xml]: why? We need it for actions

File: src/Commands/stubs/observer.stub (hunk 4 above)
Comment (masterjus) [reply src/Commands/stubs/observer.stub:7]: yes. It's in Observer make command. you could check it

File: coverage.xml (hunk 2 above)
Comment (andrew-nuwber) [reply coverage.xml]: File like that will be generated during the action, won't it?

Comment (masterjus) [reply coverage.xml]: aww! yep

**Review by mrdzen — APPROVED** [review]

File: .gitignore [hunk 7]
```diff
@@ -1,3 +1,4 @@
 .idea
 vendor
 composer.lock
+src/.DS_Store
```
Comment (andrew-nuwber) [inline .gitignore:4]: FYI

```
echo ".DS_Store" >> ~/.gitignore_global
echo "._.DS_Store" >> ~/.gitignore_global
echo "**/.DS_Store" >> ~/.gitignore_global
echo "**/._.DS_Store" >> ~/.gitignore_global
git config --global core.excludesfile ~/.gitignore_global
```

**Review by andrew-nuwber — APPROVED** [review]

**Pull request closed without merging**